src/Tasks/Microsoft.NET.Build.Tasks/DependencyContextBuilder.cs

```diff
@@ -523,8 +538,16 @@ private SingleProjectInfo GetProjectInfo(LockFileLibrary library)
                 throw new BuildErrorException(Strings.CannotFindProjectInfo, library.Name);
             }
 
-            string mainProjectDirectory = Path.GetDirectoryName(_mainProjectInfo.ProjectPath);
-            string fullProjectPath = Path.GetFullPath(Path.Combine(mainProjectDirectory, projectPath));
+            string fullProjectPath;
+            if (_mainProjectInfo != null)
+            {
+                string mainProjectDirectory = Path.GetDirectoryName(_mainProjectInfo.ProjectPath);
+                fullProjectPath = Path.GetFullPath(Path.Combine(mainProjectDirectory, projectPath));
+            }
+            else
+            {
+                fullProjectPath = Path.GetFullPath(projectPath);
```

Comment:
This logic concerns me.  I don't think this will work correctly if the project includes P2P references.

Maybe alternatively, we always create a `mainProjectInfo`, and instead have a boolean on it - `IncludeInDependencyContext` or something similar?